src/v/raft/types.h

```diff
@@ -397,6 +397,9 @@ enum class metadata_key : int8_t {
     last_applied_offset = 3,
 };
 
+// priority used to implement semi-deterministic leader election
```

Comment:
Adding a comment here but it's related to the commit message. 

> Added type describing voter priority. The priority will be used to
> implement semi-deterministic leader election mechanism to balance
> leaders across the cluster nodes.

What is non-deterministic about it? I think you mean to write a preference list.

Reply:
It is semi-deterministic as the whole election process is timer based, if some of the nodes would response slower it may not become a leader even if its priority is higher.